src/main/java/pathfinding/TextualDirections.java

```diff
@@ -47,7 +59,23 @@
 			lastAngle = tempA;
 		}
 
-		toReturn.add("You have arrived at " + path.get(0).getName());
+		//Case to print for the last item in the trimmed path
+		if(path.get(0) == totalPath.get(0)) //at destination
+		{
+			toReturn.add("You have arrived at " + path.get(0).getName());
+		} else if (path.get(0).getType() == 2) { //elevator. TODO: stairs should be related
+			Integer lastIndex = totalPath.indexOf(path.get(0));
+			toReturn.add("Take the elevator to floor " + totalPath.get(lastIndex-1).getFloor());
+		} else if (path.get(0).getType() > 5 && path.get(0).getType() < 20) //building entrance/exit
+		{
+			if(path.get(0).getBuilding().equals("00000000-0000-0000-0000-222222222222"))
```

Comment:
This should probably have a comment on it, the UUID is just a magic number here.